src/server/server_state.cc

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "server/server_state.h"
+
+#include "base/logging.h"
+#include "server/conn_context.h"
+
+namespace dfly {
+
+MonitorsRepo::MonitorInfo::MonitorInfo(ConnectionContext* conn)
+    : connection(conn), token(conn->conn_state.monitor->borrow_token),
+      thread_id(util::ProactorBase::GetIndex()) {
+}
+
+void MonitorsRepo::MonitorInfo::Send(std::string msg, std::uint32_t tid) {
+  if (tid == thread_id && connection) {
```

Comment:
I need to chat with you to understand the logic here.

Reply:
ok

Reply:
I will change this so that we would only apply this for a given thread context and we would have a counter per thread for the total number of monitors in the system so that it will retain the "share nothing" but would not require for a thread to do any job if its not required to.